### ISSUE TITLE
Document migration locking

### ISSUE THREAD

**Issue by AndriySvyryd** [dotnet/EntityFramework.Docs#4783]
### Issue or suggestion

Show how to handle the situation where the lock is abandoned (uncommon, usually SQLite after terminating the app)

**Comment by Trojaner** [comment]
cc @AndriySvyryd @maumar

**Comment by Trojaner** [comment]
@Sojaner what I mean is that this link is already used in stable released versions but its still a placeholder.

**Comment by roji** [comment]
Yes, we're aware - and this issue tracks adding the proper documentation instead of that placeholder. It's an undocumented feature.

**Comment by Trojaner** [comment]
Well that's kinda an interesting approach.

Because [as you can see](https://github.com/dotnet/EntityFramework.Docs/issues/4783#issuecomment-2994280598), this undocumented "feature" can cause obscure undocumented "errors" out of nowhere.

Like said, suddenly in this case I suddenly got an error about some timeout when attempting to place random locks on SQL Server while applying migrations via `Update-Database`. The error message refers to a link that does not really exist yet. As you have said, there is zero documentation on this and the end-user is just left guessing what they did wrong and how they can fix it. Once again, this is Microsoft.EntityFrameworkCore 9, a released stable and not a preview build. This doesn't seem up to the EntityFrameworkCore standards we are usually used to. Unless you somehow don't see a problem in this scenario.

Hence me pointing this out specifically.

**Comment by roji** [comment]
@Trojaner yes, as I've already acknowledged above, the documentation is missing on a feature/change which has been released. This is not an ideal situation - we try to document everything before or soon after a release - but it sometimes can happen, as resources and competing priorities sometimes mean that we need to work on other things.

This issue tracks correcting this - I'm not sure what else can be said, or what exactly you're looking for here.

**Comment by Sojaner** [comment]
For some reason, every time [Trojaner](https://github.com/Trojaner) is mentioned here, I get a notification stating that I have been mentioned in an issue. Does anyone know why?

<img width="1921" height="990" alt="Image" src="https://github.com/user-attachments/assets/e799a2f0-f9da-4cf9-b4a2-4d8c846f6cd8" />

**Comment by Trojaner** [comment]
Huh, thats strange. Sounds like a GitHub bug?

Sorry if off-topic.

**Comment by Sojaner** [comment]
> Huh, thats strange. Sounds like a GitHub bug?
> 
> Sorry if off-topic.

Don't mean to spam this thread, sorry, but it is actually a very weird bug. 😅

**Comment by IanKemp** [comment]
> [@Trojaner](https://github.com/Trojaner) yes, as I've already acknowledged above, the documentation is missing on a feature/change which has been released. This is not an ideal situation - we try to document everything before or soon after a release - but it sometimes can happen, as resources and competing priorities sometimes mean that we need to work on other things.
> 
> This issue tracks correcting this - I'm not sure what else can be said, or what exactly you're looking for here.

@roji it's not "not an ideal situation", but rather quite frankly unacceptable that this has not been addressed more than 18 months after this issue was raised. What's the point of having a link to documentation, if said documentation contains no mention of what that link purports to inform of? Used to be that a feature wasn't considered complete until it was documented... how far Microsoft has fallen.

And please stop insulting our intelligence with the "resources and competing priorities" excuse, because you work for one of the largest and wealthiest companies that has ever existed.

**Comment by roji** [comment]
@IanKemp I understand the frustration, but please keep your tone civil.

Microsoft may be Microsoft, but we're just a few engineers working on a specific product (EF), and we really do have a ton of stuff to do. We tend to prioritize urgent bug fixes and feature work, so documentation sometimes indeed falls behind and has gaps - we're just human beings.
